src/buildercore/config.py

```diff
@@ -81,22 +81,28 @@ class ConfigurationError(Exception):
 ROOTLOG.setLevel(logging.DEBUG) # *default* output level for all LOGs
 
 # StreamHandler sends to stderr by default
-H1 = logging.StreamHandler()
-H1.setLevel(logging.INFO) # output level for *this handler*
-H1.setFormatter(CONSOLE_FORMAT)
+CONSOLE_HANDLER = logging.StreamHandler()
+CONSOLE_HANDLER.setLevel(logging.INFO) # output level for *this handler*
+CONSOLE_HANDLER.setFormatter(CONSOLE_FORMAT)
 
 
 # FileHandler sends to a named file
-H2 = logging.FileHandler(LOG_FILE)
+FILE_HANDLER = logging.FileHandler(LOG_FILE)
 _log_level = os.environ.get('LOG_LEVEL_FILE', 'INFO')
-H2.setLevel(getattr(logging, _log_level))
-H2.setFormatter(FORMAT)
+FILE_HANDLER.setLevel(getattr(logging, _log_level))
+FILE_HANDLER.setFormatter(FORMAT)
 
-ROOTLOG.addHandler(H1)
-ROOTLOG.addHandler(H2)
+ROOTLOG.addHandler(CONSOLE_HANDLER)
+ROOTLOG.addHandler(FILE_HANDLER)
 
 LOG = logging.getLogger(__name__)
 logging.getLogger('paramiko.transport').setLevel(logging.ERROR)
+# TODO: leave on for FILE_HANDLER but not for CONSOLE_HANDLER
+# logging.getLogger('botocore.vendored').setLevel(logging.ERROR)
+
+def get_logger(name):
+    "ensures logging is setup before handing out a Logger object to use"
```

Comment:
Apparently this is not necessary in general. Is that just so that for example tests import this module which then configures the logging? I would recommend to move the setup of the logging to a method something like `configure_logging` and call it from a fixture in `conftest` and from the main entry point wherever it seems sensible (usually I would add after `if __name__ == "__main__"`).

(I only found a `get_logger` necessary in combination with Apache Beam which serialises and deserialises the pipeline)

Reply:
There is more than one entry point to cover, but logging configuration always stuck out as a global state problem. I'll try this in a separate PR to see how it may work.